samples/java_springboot/07.using-adaptive-cards/src/main/java/com/microsoft/bot/sample/usingadaptivecards/AdaptiveCardsBot.java

```diff
@@ -86,9 +86,11 @@
     }
 
     private static Attachment createAdaptiveCardAttachment(String filePath) {
+        Attachment adaptiveCardAttachment = new Attachment();
```

Comment:
There's no need to create a new instance of `Attachment` if you have one below. Just declare it here and assign value below:
```
Attachment attachment;
.
.
.
attachment = new Attachment();
```

Let's extend this to other changes as well 🙂

Reply:
We need the attachment to be initialized at the start since we need it in the resource declaration of the try(), maybe we can keep just one initialized at the start. What do you think?

Reply:
Yes, that's the idea. The instance could just be created outside the try block and then assigned inside:

```
Attachment attachment;
try (
    InputStream inputStream = attachment.getClass()
                .getClassLoader().getResourceAsStream(filePath)
) 
{
    attachment = new Attachment();
}
```